typespec-tests/tsp/flatten.tsp

```diff
@@ -33,6 +33,7 @@ model ApiVersionParameter {
 
 alias Request = {
   @visibility("read") @key name: string;
+  endpoint: string;
```

Comment:
Without the fix, this `endpoint` would not show up on method signature.

Cause is that it has same serializedName with `endpoint` on host
https://github.com/Azure/autorest.java/blob/7998c94e70361d50277866e3e59b842815e5fcc0/typespec-tests/tsp/flatten.tsp#L12-L18